src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs

```diff
@@ -102,7 +102,7 @@ public dynamic ViewBag
         [NonAction]
         public virtual ViewResult View()
         {
-            return View(viewName: null);
+            return View(viewName: ControllerContext.ActionDescriptor.ActionName);
```

Comment:
@Eilon @kichalla Those were exactly my concerns, that's the reason I split the PR into two commits. If you look at the first commit alone, you'll see that by moving the ActionName down to ControllerActionDescriptor I'm forced to do a downcast on the result executor in order to access the action name. That looks bad to me as from a layering perspective ViewResult/ViewResultExecutor should know nothing about ControllerActionDescriptor as that's an implementation detail of controllers.

After talking to @pranavkm I've figured out that we can get the action name in the same way we get the controller name and area name (see https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs#L280).

We can move https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs#L210 to https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs#L282 and throw an exception at that point (or maybe even remove if the expanders handle it gracefully)

Both methods in  https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngine.cs#L233 can be easily changed to return false or we can just add a null check to the condition.

@Eilon  @kichalla @pranavkm Thoughts?